### PR TITLE
soundwire: revert to platform devices

### PR DESCRIPTION
**Pull request by plbossart**
Since it's just too problematic to add sdw_master_device in upstream
reviews, let's use platform devices and move on. We'll come back to
sdw_master_device support later.

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>

### CODE REVIEW

**Comment by plbossart** [comment]
the code is also available in a rebased version as sdw/revert_to_platform_devices_rebase

**Comment by bardliao** [comment]
Verified playback/capture and module reload on CML.

File: drivers/soundwire/intel.c [hunk 1]
```diff
@@ -1300,66 +1300,66 @@ static int intel_init(struct sdw_intel *sdw)
...
 		return ret;
 	}
 
-	if (sdw->cdns.bus.prop.hw_disabled)
-		dev_info(&md->dev,
-			 "SoundWire master %d is disabled, will be ignored\n",
+	if (sdw->cdns.bus.prop.hw_disabled) {
+		dev_info(&pdev->dev, "SoundWire master %d is disabled, ignoring\n",
 			 sdw->cdns.bus.link_id);
+	}
```
Comment (ranj063) [inline drivers/soundwire/intel.c:1343]: @plbossart the parentheses are superfluous?

Comment (plbossart) [reply drivers/soundwire/intel.c:1343]: yes, I could drop them. Let's see what the test results are first before we look into style...

Comment (plbossart) [reply drivers/soundwire/intel.c:1343]: will do this in follow-up cleanup

**Comment by plbossart** [comment]
Intel validation results show no regression for SoundWire tests, so let's go ahead and merge.

**Pull request merged**